pkg/controller/component/utils.go

```diff
@@ -120,15 +120,14 @@ func hasHostNetworkCapability(synthesizedComp *SynthesizedComponent, compDef *ap
 	return false
 }
 
-func hasHostNetworkEnabled(annotations map[string]string, compName string) bool {
-	if annotations == nil {
-		return false
-	}
+func hasHostNetworkEnabled(annotations map[string]string, synthesizedComp *SynthesizedComponent) bool {
 	comps, ok := annotations[constant.HostNetworkAnnotationKey]
 	if !ok {
 		return false
 	}
-	return slices.Index(strings.Split(comps, ","), compName) >= 0
+
+	compsList := strings.Split(comps, ",")
+	return slices.Contains(compsList, synthesizedComp.Name) || slices.Contains(compsList, synthesizedComp.ShardingName)
```

Comment:
Refactor the conversion (from sharding to components) to be handled by the cluster controller, rather than embedding its into components.

Reply:
> Refactor the conversion (from sharding to components) to be handled by the cluster controller, rather than embedding its into components.

done